### PR TITLE
[add_kubernetes_metadata][Metricset state_namespace] Fix namespace filter

### PR DESCRIPTION
**Pull request by constanca-m**
## Proposed commit message

This PR:
1. Adds namespace filter to `add_kubernetes_metadata` watchers.
2. At metricset level, it fixes the `state_namespace` to also consider `namespace` filter like all the other namespaced resources.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

Relates to https://github.com/elastic/beats/issues/38978.

## How to test locally

Learn how to run your own filebeat [here](https://github.com/elastic/beats/tree/main/dev-tools/kubernetes). In next section, I explain my tests.

## Results

I have a cluster with two namespaces: `default` and `kube-system`.

<details>
<summary>This is an example of filestream using the processor filtered by namespace `default`.</summary>

```yaml
    filebeat.inputs:
      - type: filestream
        id: kubernetes-container-logs
        paths:
          - /var/log/containers/*.log
        parsers:
          - container: ~
        prospector:
          scanner:
            fingerprint.enabled: true
            symlinks: true
        file_identity.fingerprint: ~
        processors:
          - add_kubernetes_metadata:
              host: ${NODE_NAME}
              namespace: default
              matchers:
                - logs_path:
                    logs_path: "/var/log/containers/"
```

</details>

The documents fields annotated with data coming from the processor need to be coming from pods inside `default` namespace. Checking in discover I can see:

image

And:

image

So only pods from `default` get annotated as expected.

If I now remove the namespace from the processor and leave it like this:
```yaml
          - add_kubernetes_metadata:
              host: ${NODE_NAME}
              matchers:
                - logs_path:
                    logs_path: "/var/log/containers/"
```

I can see all documents get annotated with metadata:
image
image

### CODE REVIEW

**Comment by constanca-m** [comment]
The lint errors `specificMetaGen declared and not used (typecheck)` are annoying. I have tried to surpass it with `nolint:all`, `nolint:unused` and `nolint:typecheck` and none worked, so I am leaving them like this for now.

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -101,6 +101,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 
 *Filebeat*
 
+- Fix `namespace` filter option on `add_kubernetes_metadata` processor. {pull}39934[39934]
```
Comment (MichaelKatsoulis) [inline CHANGELOG.next.asciidoc:104]: This change affects all beats. Not just filebeat

**Review by MichaelKatsoulis — APPROVED** [review]
Thanks for that! Just address the changelog comment.

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -152,6 +153,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 
 *Metricbeat*
 
+- Fix `namespace` filter option on metricset `state_namespace` enricher. {pull}39934[39934]
```
Comment (gizas) [inline CHANGELOG.next.asciidoc:156]: This should be removed from here as it is inside 'Affecting all beats'

**Review by gizas — APPROVED** [review]

File: CHANGELOG.next.asciidoc (hunk 2 above)
Comment (constanca-m) [reply CHANGELOG.next.asciidoc:156]: https://github.com/elastic/beats/pull/39934#discussion_r1647107974, what do you think? It is not just for filebeat

**Pull request merged**